Src/FluentAssertions/Extensions/FluentDateOnlyExtensions.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Diagnostics;
+
+#if NET6_0_OR_GREATER
+
+namespace FluentAssertions.Extensions
+{
+    /// <summary>
+    /// Extension methods on <see cref="int"/> to allow for a more fluent way of specifying a <see cref="DateOnly"/>.
+    /// </summary>
+    [DebuggerNonUserCode]
+    public static class FluentDateOnlyExtensions
+    {
+        /// <summary>
+        /// Returns a new <see cref="DateOnly"/> value that is the current <see cref="TimeSpan"/> before the
+        /// specified <paramref name="sourceDateOnly"/>.
+        /// </summary>
+        public static DateOnly Before(this TimeSpan timeDifference, DateOnly sourceDateOnly)
+        {
+            var dayDifference = (int)timeDifference.TotalDays;
```

Comment:
I'm not sure about whether we should add these extension methods.

One _could_ write this piece of code, which would fail.
```cs
2.Hours().Before(dateOnly).Should().NotBe(dateOnly);
```

Reply:
I get your point.
To me it seems that trying to modify `DateOnly` with hours, minutes, etc will not change the original date (as it doesn't refer to time frames), but I see how it can be confusing.
I'll remove it.

Reply:
I get your point.
To me it seems that trying to modify DateOnly with hours, minutes, etc will not change the original date (as it doesn't refer to time frames), but I see how it can be confusing.
I'll remove it.